### PR TITLE
e2e tests: Observation source for every contract

### PR DESCRIPTION
**Pull request by gheorghestrimtu**

### CODE REVIEW

File: tests/e2e/common/common.go [hunk 1]
```diff
@@ -168,50 +168,56 @@ func DefaultOffChainConfigParamsFromNodes(nodes []client.Chainlink) (contracts.O
...
+			}
+			_, err = n.CreateTerraChain(&client.TerraChainAttributes{ChainID: "localterra"})
+			if err != nil {
+				return nil, err
+			}
+			if _, err = n.CreateTerraNode(&client.TerraNodeAttributes{
+				Name:          "terra",
+				TerraChainID:  relayConfig["chainID"],
+				TendermintURL: relayConfig["tendermintURL"],
+				FCDURL:        relayConfig["fcdURL"],
```
Comment (krebernisak) [inline tests/e2e/common/common.go:213]: `FCDURL` is not part of the node's config (represented by `TerraNodeAttributes`) anymore (for a while now), but actually a part of chains config (represented by `TerraChainAttributes`).

**Review by krebernisak — CHANGES_REQUESTED** [review]

File: tests/e2e/common/common.go (hunk 1 above)
Comment (krebernisak) [reply tests/e2e/common/common.go:213]: Here is the relevant issue @gheorghestrimtu:

-  https://github.com/smartcontractkit/chainlink-terra/issues/78

Comment (krebernisak) [reply tests/e2e/common/common.go:213]: It would be great if you can move this along as part of this effort 🙌

Comment (gheorghestrimtu) [reply tests/e2e/common/common.go:213]: Of course, no problem

File: tests/e2e/smoke/common/common.go [hunk 2]
```diff
@@ -220,13 +228,19 @@ func (m *OCRv2State) CreateJobs() {
 	m.SetAllAdapterResponsesToTheSameValue(5)
 	err := m.MockServer.SetValuePath("/juels", 1)
 	Expect(err).ShouldNot(HaveOccurred())
-	bi, err := common.CreateBridges(m.Nodes, m.MockServer)
+	ocr2Contracts := []string{}
+	for _, contract := range m.Contracts {
+		ocr2Contracts = append(ocr2Contracts, contract.OCR2.Address())
+	}
+	err = common.CreateTerraChainAndNode(m.Nodes)
+	Expect(err).ShouldNot(HaveOccurred())
```
Comment (krebernisak) [inline tests/e2e/smoke/common/common.go:236]: nit: switch the order here
```suggestion
        err = common.CreateTerraChainAndNode(m.Nodes)
	Expect(err).ShouldNot(HaveOccurred())

	ocr2Contracts := []string{}
	for _, contract := range m.Contracts {
		ocr2Contracts = append(ocr2Contracts, contract.OCR2.Address())
	}
```

**Review by krebernisak — APPROVED** [review]
LGTM!

It'd be great if you could check the consistency with Solana e2e test implementation, and do the same changes there 🙌

**Review by krebernisak — APPROVED** [review]

**Pull request merged**